File: src/zork/models/items/actions/AxeAction.java
package zork.models.items.actions;

import zork.interfaces.IUseAble;
import zork.models.entities.Player;

/**
 * Action for axes.
 *
 * @author d.peters
 * @version 1.0
 */
public class AxeAction implements IUseAble {
  /**
   * Use method without any param if the object should not do something with any object.
   */
  @Override
  public void use() {
    System.out.println("You swing the Axe around you. You almost killed yourself in the process.");
  }

  /**
   * Use item on the player.
   *
   * @param player the player object.
   */
  @Override
  public void use(Player player) {
    use();
  }
}

File: src/zork/models/items/actions/AxeAction.kt
package zork.models.items.actions;

import zork.interfaces.IItemAction

/**
 * Action for axes.
 *
 * @author d.peters
 * @version 1.0
 */
 class AxeAction: IItemAction {
  /**
   * Use method without any param if the object should not do something with any object.
   */
  override fun execute () {
    System.out.println("You swing the Axe around you. You almost killed yourself in the process.")
  }
}
